File: app/src/main/java/com/mbcdev/goddamn/GodDamnService.java
package com.mbcdev.goddamn;

import android.media.MediaPlayer;
import android.os.Build;
import android.service.quicksettings.TileService;
import android.support.annotation.RequiresApi;

/**
 * Just stop judging me. It's your fault Egan and Brenda.
 *
 * Created by bcarroll on 13/11/2017.
 */
@RequiresApi(api = Build.VERSION_CODES.N)
public class GodDamnService extends TileService {

    @Override
    public void onClick() {
        final MediaPlayer mediaPlayer = MediaPlayer.create(GodDamnService.this, R.raw.goddamn);
        mediaPlayer.start();
    }
}

File: app/src/main/kotlin/goddamn/GodDamnService.kt
package goddamn

import android.os.Build
import android.service.quicksettings.TileService
import android.support.annotation.RequiresApi

/**
 * Just stop judging me. It's your fault Egan and Brenda.
 *
 * It wasn't my fault before, but some of it is now.
 *
 * Created by bcarroll on 13/11/2017.
 */
@RequiresApi(api = Build.VERSION_CODES.N)
class GodDamnService : TileService() {

    override fun onClick() {
        GodDamnApp.get(this).godDamn()
    }
}
